### PR TITLE
Netsuite base component

### PR DESCRIPTION
**Pull request by NileshPant1999**

### CODE REVIEW

File: src/app/core/models/enum/enum.model.ts [hunk 1]
```diff
@@ -156,6 +157,16 @@ export enum QBOOnboardingState {
   CLONE_SETTINGS = 'CLONE_SETTINGS'
 }
 
+export enum NetsuiteOnboardingState {
+  CONNECTION = 'CONNECTION',
+  SUBSIDIARY_MAPPING = 'SUBSIDIARY_MAPPING',
+  MAP_EMPLOYEES = 'MAP_EMPLOYEES',
```
Comment (NileshPant1999) [inline src/app/core/models/enum/enum.model.ts:163]: will remove this

File: src/app/core/models/enum/enum.model.ts [hunk 2]
```diff
@@ -30,7 +30,8 @@ export enum InAppIntegration {
   TRAVELPERK = 'TravelPerk',
   INTACCT = 'Sage Intacct',
   SAGE300 = 'Sage 300 CRE',
-  BUSINESS_CENTRAL = 'Dynamics 365 Business Central'
+  BUSINESS_CENTRAL = 'Dynamics 365 Business Central',
+  NETSUITE = 'Netsuite'
```
Comment (ashwin1111) [inline src/app/core/models/enum/enum.model.ts:34]: `NetSuite`

File: src/app/core/models/enum/enum.model.ts (hunk 1 above)
Comment (ashwin1111) [inline src/app/core/models/enum/enum.model.ts:163]: +1

File: src/app/core/models/netsuite/db/netsuite-workspace.model.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { NetsuiteOnboardingState } from "../../enum/enum.model";
+import { Workspace } from "../../db/workspaces.model";
+
+export interface NetsuiteWorkspace extends Workspace {
+    fyle_org_id: string;
+    fyle_currency: string;
+    qbo_realm_id: string;
```
Comment (ashwin1111) [inline src/app/core/models/netsuite/db/netsuite-workspace.model.ts:7]: remove

File: src/app/integrations/landing/landing.component.html [hunk 4]
```diff
@@ -81,6 +81,14 @@
                         </span>
                     </div> -->
                 </div>
+                <div>
+                    <div class="landing--accounting-app tw-mr-20-px" (click)="openAccountingIntegrationApp(AccountingIntegrationApp.NETSUITE)">
```
Comment (ashwin1111) [inline src/app/integrations/landing/landing.component.html:85]: we do have netsuite card already, why is this required?

File: src/app/integrations/netsuite/netsuite.component.ts [hunk 5]
```diff
@@ -7,9 +17,68 @@ import { Component, OnInit } from '@angular/core';
...
+  ) {
+    this.windowReference = this.windowService.nativeWindow;
+  }
+
+  private navigate(): void {
+    const pathName = this.windowReference.location.pathname;
+    if (pathName === '/integrations/netsuite') {
+      const onboardingStateComponentMap = {
+        [NetsuiteOnboardingState.CONNECTION]: '/integrations/netsuite/onboarding/landing',
+        [NetsuiteOnboardingState.SUBSIDIARY_MAPPING]: '/integrations/netsuite/onboarding/subsiduary-mappings',
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite.component.ts:45]: typo

File: src/app/integrations/netsuite/netsuite.component.ts [hunk 6]
```diff
@@ -7,9 +17,68 @@ import { Component, OnInit } from '@angular/core';
...
+        [NetsuiteOnboardingState.MAP_EMPLOYEES]: '/integrations/netsuite/onboarding/employee_settings',
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite.component.ts:46]: not needed

File: src/app/core/models/enum/enum.model.ts [hunk 7]
```diff
@@ -160,8 +160,6 @@ export enum QBOOnboardingState {
 
 export enum NetsuiteOnboardingState {
   CONNECTION = 'CONNECTION',
-  SUBSIDIARY_MAPPING = 'SUBSIDIARY_MAPPING',
```
Comment (ashwin1111) [inline src/app/core/models/enum/enum.model.ts]: SUBSIDIARY_MAPPING is required, it's a separate step

File: src/app/integrations/netsuite/netsuite.component.ts [hunk 8]
```diff
@@ -42,8 +42,6 @@ export class NetsuiteComponent implements OnInit {
     if (pathName === '/integrations/netsuite') {
       const onboardingStateComponentMap = {
         [NetsuiteOnboardingState.CONNECTION]: '/integrations/netsuite/onboarding/landing',
-        [NetsuiteOnboardingState.SUBSIDIARY_MAPPING]: '/integrations/netsuite/onboarding/subsiduary-mappings',
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite.component.ts]: this is required

**Review by ashwin1111 — APPROVED** [review]
